src/utils/daos_hdlr.c

```diff
@@ -2767,56 +2753,57 @@ dm_disconnect(struct cmd_args_s *ap,
 	      struct file_dfs *src_file_dfs,
 	      struct file_dfs *dst_file_dfs)
 {
+	 /* The fault injection tests expect no memory leaks but inject faults that
+	 * block umount/close/disconnect calls, etc. So, if I use GOTO and return the error
+	 * code immediately after a failure to umount/close/disconnect then fault injection
+	 * will always report a memory leak. Is it better to immediately return if one of
+	 * these fails? This will cause memory leaks in fault injection tests for fs copy,
+	 * so not sure what is the best thing to do here. 
+	 */
```

Comment:
Similar to the conversation between @mchaarawi and @ashleypittman. At this time, I'm not sure what the best solution is.